tests/conformance/pubsub/pubsub.go

```diff
@@ -137,14 +179,17 @@ func ConformanceTests(t *testing.T, props map[string]string, ps pubsub.PubSub, c
 		// So, wait for some time here.
 		time.Sleep(config.waitDurationToPublish)
 		t.Run("publish", func(t *testing.T) {
-			for k := 0; k < config.messageCount; k++ {
-				data := []byte("message-" + strconv.Itoa(k))
+			for k := 1; k <= config.messageCount; k++ {
+				data := []byte(fmt.Sprintf("%s%d", dataPrefix, k))
 				err := ps.Publish(&pubsub.PublishRequest{
 					Data:       data,
 					PubsubName: config.pubsubName,
 					Topic:      config.testTopicName,
 					Metadata:   config.publishMetadata,
 				})
+				if err == nil {
+					awaitingMessages[string(data)] = struct{}{}
```

Comment:
use sets instead of this?

Reply:
In Go, `map[string]struct{}` is a set.

Reply:
I'd like to stick with simple Go maps as sets instead of using `k8s.io/apimachinery/pkg/util/sets` as this use case is simple.